low-power-digital-signage/firmware/README.md

```diff
@@ -0,0 +1,66 @@
+the prerequisites below installed.
+
+## Prerequisites
+
+1. Download and install [Visual Studio Code](https://code.visualstudio.com/).
+1. Install the [PlatformIO IDE extension](https://marketplace.visualstudio.com/items?itemName=platformio.platformio-ide)
+via the Extensions menu of Visual Studio Code.
+![Visual Studio Code extensions menu with a search of "platformio"](../assets/images/platformio-extension.png)
+
+### Opening
+
+Once you have Visual Studio Code and the PlatformIO extension installed, you next need
+to open the firmware within PlatformIO.
+
+1. Download or clone this repository, so that you have a copy of the firmware source
+code locally.
+1. Open the PlatformIO extension by clicking on the PlatformIO logo in the menu bar. Next,
+click the “Open” option under the “PIO Home” menu  and finally “Open Project”.
+![Instructions on how to open a project in PlatformIO](../assets/images/platformio-open-project.png)
+1. Select the `firmware` folder from within your local copy of this repository,
+and click the **Open "firmware"** button.
+![How to open the firmware folder in PlatformIO](../assets/images/platformio-open-firmware.png)
+
+### Flashing
+
+With the firmware project open, you can now make any changes you’d like to the firmware
+source code, and then flash the firmware to your device.
+
+Complete the following steps to upload firmware in in PlatformIO.
+
+1. Connect your Swan to your computer using [these steps](https://dev.blues.io/quickstart/swan-quickstart/#programming-swan-platformio).
+
+> **NOTE**: The project’s default configuration assumes you’re using an
+[STLINK-V3MINI programmer and debugger](https://shop.blues.io/products/stlink-v3mini).
+If you’re not, make sure to open your `platformio.ini` file and change the `upload_protocol`
+to `dfu`.
+
+2. Once you have everything connected, click the checkmark button that appears at the
+bottom of your Visual Studio Code window.
+![Uploading firmware in PlatformIO](../assets/images/platformio-upload.png)
+
+Once the firmware is running on your device, you might additionally want to open a serial
+monitor to view the firmware’s logs. You can do so by clicking the plug button that appears
+at the bottom of your Visual Studio Code window.
+
+![Opening a serial monitor in PlatformIO](../assets/images/platformio-serial-monitor.png)
+
+The firmware operates on two environment variables, which you can set at the fleet or device level using either the Notehub UI or one of the [shell scripts in the `scripts` folder](../scripts/).
+
+1. `display_values` which is a semicolon-delimited string of text to display and/or bitmap images to load from an SD card to the eInk display. If this variable has a single value (either text or an image name), that will be loaded. Otherwise, the values are loaded into an array and the host will display one item ata time using the `display_interval_sec` variable or a default value, if not set.
```

Comment:
```suggestion
1. `display_values` which is a semicolon-delimited string of text to display and/or bitmap images to load from an SD card to the eInk display. If this variable has a single value (either text or an image name), that will be loaded. Otherwise, the values are loaded into an array and the host will display one item ata time using the `display_interval_sec` variable or a default value, if not set.
```
```suggestion
1. `display_values` which is a semicolon-delimited string of text to display and/or bitmap images to load from an SD card to the e-ink display. If this variable has a single value (either text or an image name), that will be loaded. Otherwise, the values are loaded into an array and the host will display one item at a time using the `display_interval_sec` variable or a default value, if not set.
```